extensions/include/extensions-export.h

```diff
@@ -0,0 +1,57 @@
+#if __has_include("cmark-gfm-extensions_export.h")
```

Comment:
`__has_include` is not a standard extension, please add a fallback.

```suggestion
#if !defined(__has_include)
#define __has_include(include) 0
#endif

#if __has_include("cmark-gfm-extensions_export.h")
```

Reply:
Should i try to find an alternative for the section in `mutex.h` where it tries to find and include `unistd.h`?

https://github.com/apple/swift-cmark/blob/3780d733061ba47d1b9dfc4e224079e53d3f3023/src/include/mutex.h#L8-L10

That section already compiles just fine with Clang, GCC, and MSVC, in the situations that i've tried. If we're trying to avoid `__has_include` here, i imagine we should provide this fallback for the entire project.

Reply:
It seems that `__has_include` was standardized in [C++17](https://en.cppreference.com/w/cpp/17) - it's valid if this is too new of a target to restrict ourselves to, but it is standard now.

Reply:
Yes, it is in C++17, however, the project is a C codebase, and does not follow the C++ standards but rather the C standards.  I think that providing a fallback is a good idea.

Reply:
I've pushed some commits that scrub out the uses of `__has_include` in the repo, leaving behind one in the prebuilt config.h that's behind a check.